src/lib.rs

```diff
@@ -202,6 +309,86 @@ impl<T: Transport + Sized> SatsCard<T> {
         response
     }
 
+    pub fn derive(&mut self) -> Result<DeriveResponse, Error> {
+        let cmd = DeriveCommand::for_satscard(self.card_nonce.clone());
+        let resp: Result<DeriveResponse, Error> = self.transport.transmit(cmd);
+        if let Ok(derive_resp) = &resp {
+            self.card_nonce = derive_resp.card_nonce.clone();
+
+            // TODO: verify reponse
+            // The digest:
+                // b'OPENDIME' (8 bytes)    
+                // (card_nonce - 16 bytes)
+                // (nonce from command - 16 bytes)
+                // (chain_code - 32 bytes)
+            // must be signed by the master_pubkey
+        }
+        resp
+    }
+
+    pub fn certs_check(&mut self, nonce: Vec<u8>) -> Result<String, Error> {
+        let card_nonce = self.card_nonce.clone();
+
+        let certs_cmd = CertsCommand::default();
+        let certs_response: CertsResponse = self.transport.transmit(certs_cmd)?;
+
+        let check_cmd = CheckCommand::new(nonce.clone());
+        let check_response: Result<CheckResponse, Error> = self.transport.transmit(check_cmd);
+        if let Ok(response) = &check_response {
+            self.card_nonce = response.card_nonce.clone();
+        }
+
+        assert!(self.verify_satscard_signature(
+            check_response.unwrap().auth_sig.clone(), 
+            card_nonce, 
+            nonce.clone(), 
+        ).is_ok());
+
+        let mut pubkey = self.pubkey.clone();
+        for sig in &certs_response.cert_chain() {
+            // BIP-137: https://github.com/bitcoin/bips/blob/master/bip-0137.mediawiki
+            let subtract_by = match sig[0] {
+                27..=30 => 27, // P2PKH uncompressed
+                31..=34 => 31, // P2PKH compressed
+                35..=38 => 35, // Segwit P2SH
+                39..=42 => 39, // Segwit Bech32
+                _ => panic!("Unrecognized BIP-137 address")
+            };
+            let rec_id = RecoveryId::from_i32((sig[0] as i32) - subtract_by).unwrap();
+            let (_, sig) = sig.split_at(1);
+            let rec_sig = RecoverableSignature::from_compact(sig, rec_id).unwrap();
+            let md = Message::from_hashed_data::<sha256::Hash>(pubkey.serialize().as_slice());            
+            pubkey = self.secp.recover_ecdsa(&md, &rec_sig).unwrap();
+        }
+
+        // todo - as a constant?
+        let factory_root_keys: Vec<(Vec<u8>, String)>= vec!(
+            (from_hex("03028a0e89e70d0ec0d932053a89ab1da7d9182bdc6d2f03e706ee99517d05d9e1"), "Root Factory Certificate".to_string()), 
+            (from_hex("027722ef208e681bac05f1b4b3cc478d6bf353ac9a09ff0c843430138f65c27bab"), "Root Factory Certificate (TESTING ONLY)".to_string())
+        );
+        if let Some((_, value)) = factory_root_keys.iter().find(|(k, _)| *k == pubkey.serialize()) {
+            Ok(value.to_string())
+        } else {
+            Err(Error::CiborValue("Root cert is not from Coinkite. Card is counterfeit.".to_string()))
+        }
```

Comment:
If we go with a `FactoryRootKey` the end of this function becomes:
```suggestion
       pubkey.try_into()
```
And returns `Result<FactoryRootKey,Error>`.  But as mentioned above we don't need to decide for this PR, so unless you love my `Enum` idea we can leave this as is for now.